muxer/mplex/muxed_stream.py

```diff
@@ -1,34 +1,113 @@
+import asyncio
 from .muxed_stream_interface import IMuxedStream
+from .constants import HEADER_TAGS
+
 
 class MuxedStream(IMuxedStream):
     """
     reference: https://github.com/libp2p/go-mplex/blob/master/stream.go
     """
 
-    def __init__(self, protocol_id, stream_name):
-        self.protocol_id = protocol_id
-        self.name = stream_name
+    def __init__(self, stream_id, initiator, muxed_conn):
+        """
+        create new MuxedStream in muxer
+        :param stream_id: stream stream id
+        :param initiator: boolean if this is an initiator
+        :param muxed_conn: muxed connection of this muxed_stream
+        """
+        self.id = stream_id
+        self.initiator = initiator
+        self.muxed_conn = muxed_conn
+
+        # self.read_deadline = None
+        # self.write_deadline = None
+
+        self.local_closed = False
+        self.remote_closed = False
+
+    def get_flag(self, action):
+        """
+        get header flag based on action for mplex
+        :param action: action type in str
+        :return: int flag
+        """
+        if self.initiator:
+            return HEADER_TAGS[action]
+        else:
+            return HEADER_TAGS[action] - 1
 
     def read(self):
-        pass
+        """
+        read from stream til end of file
```

Comment:
Can you update the comment to say "read any messages from buffer associate with stream"